src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,92 @@
+    private final RelativeEncoder pivotEncoder = pivotMotor.getEncoder();
+    private final AbsoluteEncoder pivotAbsoluteEncoder = pivotMotor.getAbsoluteEncoder(Type.kDutyCycle);
+    private final TalonFX intakeTalonFront = new TalonFX(RobotMap.INTAKE_TALON_FRONT_CAN_ID);
+    private final Follower intakeTalonBack = new Follower(intakeTalonFront.getDeviceID(), false); // TODO check
+    private ProfiledPIDController pivotController = new ProfiledPIDController(IntakeCal.INTAKE_PIVOT_P,
+            IntakeCal.INTAKE_PIVOT_I, IntakeCal.INTAKE_PIVOT_D, new TrapezoidProfile.Constraints(
+                    IntakeCal.PIVOT_MAX_VELOCITY_DEG_PER_SECOND,
+                    IntakeCal.PIVOT_MAX_ACCELERATION_DEG_PER_SECOND_SQUARED));
+
+    public enum IntakePosition {
+        DEPLOYED,
+        STOWED,
+        SAFE
+    }
+
+    private TreeMap<IntakePosition, Double> intakePositionMap;
+    private IntakePosition latestPosition = IntakePosition.STOWED;
+    private IntakePosition desiredPosition = IntakePosition.STOWED;
+
+    public Intake() {
+        intakePositionMap = new TreeMap<IntakePosition, Double>();
+        intakePositionMap.put(IntakePosition.DEPLOYED, IntakeCal.INTAKE_DEPLOYED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.STOWED, IntakeCal.INTAKE_STOWED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.SAFE, IntakeCal.INTAKE_SAFE_POSITION_DEGREES);
+    }
+
+    /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+    public double getCosineArmAngle() {
+        return Math.cos(
+                Units.degreesToRadians(
+                        pivotEncoder.getPosition() - IntakeConstants.INTAKE_POSITION_WHEN_HORIZONTAL_DEGREES));
+    }
+
+    public void moveToPos(IntakePosition pos) {
+        pivotController.setGoal(intakePositionMap.get(pos));
+        double intakeDemandVoltsA = pivotController.calculate(pivotEncoder.getPosition());
+        double intakeDemandVoltsB = IntakeCal.INTAKE_PIVOT_FEEDFORWARD
+                .calculate(pivotController.getSetpoint().velocity);
+        double intakeDemandVoltsC = IntakeCal.ARBITRARY_INTAKE_PIVOT_FEEDFORWARD_VOLTS * getCosineArmAngle();
+        pivotMotor.setVoltage(intakeDemandVoltsA + intakeDemandVoltsB + intakeDemandVoltsC);
+
+        if (pos == IntakePosition.DEPLOYED) {
+            startRollers();
+        } else {
+            stopRollers();
+        }
+    }
+
+    public boolean atDesiredIntakePosition(IntakePosition pos) {
+        double desiredPositionDegrees = intakePositionMap.get(pos);
+        double intakePositionDegrees = pivotEncoder.getPosition();
+        return Math.abs(intakePositionDegrees - desiredPositionDegrees) <= IntakeCal.INTAKE_MARGIN_DEGREES;
+    }
+
+    public boolean clearOfConveyorZone() {
+        return pivotEncoder.getPosition() > IntakeCal.CONVEYOR_ZONE_THRESHOLD_DEGREES;
```

Comment:
Whether this is > or < depends on what 0deg is; have we defined that?

Reply:
We should define and write that somewhere yeah

Reply:
90 degrees now represents when the intake is at the shooter direction, therefore this should still be a >